internal/provider/indices_resource.go

```diff
@@ -399,13 +523,40 @@ func (r *indicesResource) Read(ctx context.Context, req resource.ReadRequest, re
 		if newState.Settings.Type.ValueString() == "structured" {
 			newState.Settings.FilterStringMaxLength = types.Int64Null()
 			newState.Settings.TreatUrlsAndPointersAsImages = types.BoolNull()
+			newState.Settings.TreatUrlsAndPointersAsMedia = types.BoolNull()
 		}
 
 		// Handle image_preprocessing.patch_method
 		if newState.Settings.ImagePreprocessing.PatchMethod.ValueString() == "" {
 			newState.Settings.ImagePreprocessing.PatchMethod = types.StringNull()
 		}
 
+		// Handle video and audio preprocessing fields
+		if newState.Settings.VideoPreprocessing != nil &&
+			newState.Settings.VideoPreprocessing.SplitLength.ValueInt64() == 0 &&
+			newState.Settings.VideoPreprocessing.SplitOverlap.ValueInt64() == 0 {
+			newState.Settings.VideoPreprocessing = nil
+		}
+
+		if newState.Settings.AudioPreprocessing != nil &&
+			newState.Settings.AudioPreprocessing.SplitLength.ValueInt64() == 0 &&
+			newState.Settings.AudioPreprocessing.SplitOverlap.ValueInt64() == 0 {
+			newState.Settings.AudioPreprocessing = nil
+		}
+
+		// Handle model properties
+		if newState.Settings.ModelProperties != nil {
+			if newState.Settings.ModelProperties.Name.ValueString() == "" &&
```

Comment:
This logic is different to above where you check `IsNull`. Should they be the same? If not, why not? Again feels like it could be a method on the struct to make it more obvious what you're actually checking.

Reply:
I recently changed the resource to refer to objects through a pointer (*ModelPropertiesModelCreate), whereas in datasource it remains a direct struct field. 

I couldn't complete the transition due to issues with testing in staging the past few days, but it seems to be working now and I'm working through the bugs!